development/lib/get-manifest-flag.ts

```diff
@@ -0,0 +1,102 @@
+import path from 'node:path';
+import fs from 'node:fs/promises';
+import { promisify } from 'node:util';
+import { exec as callbackExec } from 'node:child_process';
+
+import { hasProperty } from '@metamask/utils';
+import { merge } from 'lodash';
+
+import type { ManifestFlags } from '../../app/scripts/lib/manifestFlags';
+
+const exec = promisify(callbackExec);
+const PR_BODY_FILEPATH = path.resolve(
+  __dirname,
+  '..',
+  '..',
+  'changed-files',
+  'pr-body.txt',
+);
+
+/**
+ * Search a string for `flags = {...}` and return ManifestFlags if it exists
+ *
+ * @param str - The string to search
+ * @param errorType - The type of error to log if parsing fails
+ * @returns The ManifestFlags object if valid, otherwise undefined
+ */
+function regexSearchForFlags(str: string, errorType: string): ManifestFlags {
+  // Search str for `flags = {...}`
+  const flagsMatch = str.match(/flags\s*=\s*(\{.*\})/u);
+
+  if (flagsMatch) {
+    try {
+      // Get 1st capturing group from regex
+      return JSON.parse(flagsMatch[1]);
+    } catch (error) {
+      console.error(
+        `Error parsing flags from ${errorType}, ignoring flags\n`,
+        error,
+      );
+    }
+  }
+
+  return {};
+}
+
+/**
+ * Get flags from the GitHub PR body if they are set
+ *
+ * To use this feature, add a line to your PR body like:
+ * `flags = {"sentry": {"tracesSampleRate": 0.1}}`
+ * (must be valid JSON)
+ *
+ * @returns Any manifest flags found in the PR body
+ */
+async function getFlagsFromPrBody(): Promise<ManifestFlags> {
+  let body: string;
+  try {
+    body = await fs.readFile(PR_BODY_FILEPATH, 'utf8');
+  } catch (error) {
+    if (
+      error instanceof Error &&
+      hasProperty(error, 'code') &&
+      error.code === 'ENOENT'
+    ) {
+      console.debug('No pr-body.txt, ignoring flags');
+      return {};
+    }
+    throw error;
+  }
+
+  return regexSearchForFlags(body, 'PR body');
+}
+
+/**
+ * Get flags from the Git message if they are set
+ *
+ * To use this feature, add a line to your commit message like:
+ * `flags = {"sentry": {"tracesSampleRate": 0.1}}`
+ * (must be valid JSON)
+ *
+ * @returns Any manifest flags found in the commit message
+ */
+async function getFlagsFromGitMessage(): Promise<ManifestFlags> {
+  const gitMessage = (await exec(`git show --format='%B' --no-patch "HEAD"`))
+    .stdout;
+
+  return regexSearchForFlags(gitMessage, 'git message');
+}
+
+/**
+ * Get any manifest flags found in the PR body and git message.
+ *
+ * @returns Any manifest flags found
+ */
+export async function getManifestFlags(): Promise<ManifestFlags> {
```

Comment:
We now have two functions in the code base named `getManifestFlags` that both end up getting the same flags, but in different ways (runtime vs compile time) :-/ One is async (this one), and the other must stay synchronous as it is designed to be blocking and used on app initialization.

Think you could rename one or the other?

Reply:
Good point. Seems better to rename this one. Maybe `fetchManifestFlags` or `createManifestFlags`?

Reply:
Either is fine with me!

Reply:
@Gudahtt @davidmurdoch @DDDDDanica I think this one should be called `fetchManifestFlagsFromGitAndPR`